airflow/providers/openlineage/sqlparser.py

```diff
@@ -0,0 +1,297 @@
+        hook: BaseHook,
+        database_info: DatabaseInfo,
+        database: str | None = None,
+    ) -> OperatorLineage:
+        """
+        Parses SQL statement(s) and generate OpenLineage metadata:
+            * input tables with schemas parsed
+            * output tables with schemas parsed
+            * run facets
+            * job facets
+
+        :param sql: a SQL statement or list of SQL statement to be parsed
+        :param hook: Airflow Hook used to connect to the database
+        :param database_info: database specific information
+        :param database: when passed it takes precedence over parsed database name
+        """
+        job_facets = {"sql": SqlJobFacet(query=SQLParser.normalize_sql(sql))}
+
+        parse_result: SqlMeta | None = self.parse(SQLParser.split_sql_string(sql))
+        if not parse_result:
+            return OperatorLineage(job_facets=job_facets)
+
+        run_facets: dict = {}
+
+        if parse_result.errors:
+            run_facets["extractionError"] = ExtractionErrorRunFacet(
+                totalTasks=len(sql) if isinstance(sql, list) else 1,
+                failedTasks=len(parse_result.errors),
+                errors=[
+                    ExtractionError(
+                        errorMessage=error.message,
+                        stackTrace=None,
+                        task=error.origin_statement,
+                        taskNumber=error.index,
+                    )
+                    for error in parse_result.errors
+                ],
+            )
+
+        namespace = SQLParser.create_namespace(database_info=database_info)
+        inputs, outputs = self.parse_table_schemas(
+            hook=hook,
+            inputs=parse_result.in_tables,
+            outputs=parse_result.out_tables,
+            namespace=namespace,
+            database=database,
+            database_info=database_info,
+        )
+
+        return OperatorLineage(
+            inputs=inputs,
+            outputs=outputs,
+            run_facets=run_facets,
+            job_facets=job_facets,
+        )
+
+    @staticmethod
+    def create_namespace(database_info: DatabaseInfo) -> str:
+        return (
+            f"{database_info.scheme}://{database_info.authority}"
+            if database_info.authority
+            else database_info.scheme
+        )
+
+    @staticmethod
+    def normalize_sql(sql: list[str] | str) -> str:
+        """Makes sure to return a semicolon-separated SQL statements"""
+        return ";\n".join(stmt.rstrip(" ;\r\n") for stmt in SQLParser.split_sql_string(sql))
+
+    @staticmethod
+    def split_sql_string(sql: list[str] | str) -> list[str]:
+        """
+        Split SQL string into list of statements
+        Tries to use `DbApiHook.split_sql_string` if available.
+        Otherwise, uses the same logic.
+        """
+        split_statement: Callable[[str], list[str]]
+        try:
+            from airflow.providers.common.sql.hooks.sql import DbApiHook
```

Comment:
I think SQL provider is one we can hard require. Also, will we ever get called here if SQL provider does not exist?

Reply:
It's hard to predict where else SQLParser could be used but I can imagine that could happen without use of `common.sql` provider. That's why I've created the logic to reflect the same behaviour even when no `common.sql` provider is available.